lib/libthr/thread/thr_init.c

```diff
@@ -456,6 +456,18 @@ init_main_thread(struct pthread *thread)
 bool
 __thr_get_main_stack_base(char **base)
 {
+#ifdef __CHERI_PURE_CAPABILITY__
```

Comment:
I guess one risk here is that I need to verify this function is always called by the main thread.

Upon further review, it is (other use case is for the map_stacks_exec hook when single-threaded).

However, there is another copy of this code in libc/gen/elf_utils.c (_libc_map_stacks_exec) that probably needs the same treatment as this of using cheri_getstack().  OTOH, it might be that all this notion of mapping stacks executable is not relevant on purecap (in which case we should maybe #ifdef out all the map_stacks_exec crud for purecap).

Reply:
I believe there is no way to map the main stack executable under CheriABI so code to do so is broken (or perhaps a no-op if return values aren't checked).